tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper.py

```diff
@@ -326,11 +326,11 @@ def trainable(self, value):
 
   @property
   def trainable_weights(self):
-    return self.layer.trainable_weights + self._trainable_weights
```

Comment:
Maybe I've misunderstood but self._trainable_weights and self._non_trainable_weights should be updated automatically during calls to self.add_weight(). 

By removing "self._trainable_weights", someone calling layer.trainable_weights will no longer see all the weights that your wrapper is adding, which breaks the interface. 

You can try creating a unit test that checks what layer.trainable_weights outputs and see if it makes sense and look at the base tf.keras.layers.Layer implementation to see what it does.